src/main/java/MiniMax.java

```diff
@@ -1,5 +1,94 @@
 public class MiniMax {
-    public int treeScore(Node node) {
-        return node.getScore();
+    private final Board board;
+    private Rules rules = new Rules();
+    private String maximizer;
+    private String minimizer;
+
+    public MiniMax(Board board) {
+        this.board = board;
+    }
+
+    private Integer evalBoard(int depth) {
+        final int WINNING_SCORE = 10;
+        final int LOSING_SCORE = -10;
+        final int DRAW_SCORE = 0;
+        if (rules.victoryCheck(board.currentMoves(), maximizer).equals(maximizer)){
+            return WINNING_SCORE - depth;
+        }
+        if (rules.victoryCheck(board.currentMoves(), minimizer).equals(minimizer)) {
+            return LOSING_SCORE + depth;
+        }
+        return DRAW_SCORE;
+    }
+
+    private Integer miniMax(int depth, boolean isMax) {
+        int score = evalBoard(depth);
+
+        if (score != 0) {
+            return score;
+        }
+
+        if (board.availableSpaces().isEmpty()) {
+            return 0;
+        }
+        if (isMax) {
+            int best = Integer.MIN_VALUE;
+
+            for (int square : board.availableSpaces()) {
+                board.markBoard(square + 1, maximizer);
+                best = Integer.max(best, miniMax(depth +1, false));
```

Comment:
minor clean  up: need a space between `+` and `1`

Reply:
also, i think you can do `!isMax` in `miniMax(depth + 1, !isMax)`. do you think that reads better?

Reply:
Spacing fixed, when I attempt to switch to isMax and !isMax IntelliJ prompts me to turn it back to true/false, gives warning.